src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -165,22 +185,77 @@ class TaskDeploymentService {
     }
 
     /**
-     * Helper method to initialise the lateinit vars [taskAssumeRoleString] and [taskRolePolicyString] by
-     * converting the associated `@Value` parameters to Strings and replacing `ADDITIONAL_PERMISSIONS` in
-     * [taskRolePolicyString] with the provided [additionalPermissions]
+     * Helper method to initialise the lateinit vars [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] by
+     * converting the JSON to an instance of `JsonObject` data class and adding any requirements before serialising back to JSON
      *
-     * @return [Pair] of [taskRolePolicyString] to [taskAssumeRoleString] for ease of access.
+     * @return [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] respectively.
      */
-    fun parsePolicyDocuments(additionalPermissions: List<String>): Pair<String, String> {
-        var replaceString = ""
-        if (additionalPermissions.isNotEmpty()) {
-            logger.info("Adding permissions to containers", "permissions" to additionalPermissions.joinToString())
-            val permissionsJson = additionalPermissions.joinToString(prefix = "\"", separator = "\",\"", postfix = "\"")
-            replaceString = "$permissionsJson,"
+    fun parsePolicyDocument(resource: Resource, sidAndAdditions: Map<String, List<String>>, key: String): String{
+        val mapper = ObjectMapper()
+                .setPropertyNamingStrategy(AwsPropertyNamingStrategy())
+        val obj = mapper.readValue(resource.url, AwsIamPolicyJsonObject::class.java)
+        obj.Statement.forEach {statement ->
+            sidAndAdditions.forEach {
+                if(it.key == statement.Sid) {
+                    if (key == "Resource") statement.Resource = statement.Resource.plus(it.value)
+                    else if (key == "Action") statement.Action = statement.Action.plus(it.value)
+                    else throw IllegalArgumentException("Key does not match expected values: \"Resource\" or \"Action\"")
+                }
+            }
+        }
+        return mapper.writeValueAsString(obj)
+    }
+
+/**
+ * Helper method to parse user access details into ARN format - returns list of these and the JupyterBucket ARN
+ */
+    fun createArnStringsList(pathPrefix: List<String>, pathSuffix: String, jupyterBucketArn: String): List<String>{
+        var returnList = listOf(jupyterBucketArn)
+        pathPrefix.forEach{
+            returnList = returnList.plus("arn:aws:kms:${configurationResolver.awsRegion}:${awsCommunicator.getAccNumber()}:alias/${it}-${pathSuffix}")
```

Comment:
Same point about avoiding mutations
```suggestion
    fun createArnStringsList(pathPrefix: List<String>, pathSuffix: String, jupyterBucketArn: String): List<String>{
        val kmsArns = pathPrefix.map { "arn:aws:kms:${configurationResolver.awsRegion}:${awsCommunicator.getAccNumber()}:alias/${it}-${pathSuffix}" }
        return kmsArns.plus(jupyterBucketArn)
    }
```